### PR TITLE
feat: extend session search service

### LINKED ISSUE #6029: Extend search user sessions
As a developer I want to be able to show all sessions of a user on his profile, so he is able to terminate certain sessions that are not used anymore. 

**Acceptance Criteria**
- [x] I can Search Sessions with a user id filter
- [x] I can filter the sessions per creation date
- [x] I can order the sessions per creation date

### PR DESCRIPTION
**Pull request by sp132**
- add two more searching criteria - human user id and session creation date
- optional sorting by the session creation date
- align receiver names

Closes #6029 

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/query/session.go [hunk 1]
```diff
@@ -257,6 +258,14 @@ func NewSessionCreatorSearchQuery(creator string) (SearchQuery, error) {
 	return NewTextQuery(SessionColumnCreator, creator, TextEquals)
 }
 
+func NewUserIDSearchQuery(id string) (SearchQuery, error) {
+	return NewTextQuery(HumanUserIDCol, id, TextEquals)
```
Comment (stebenz) [inline internal/query/session.go:262]: Shouldn't this be with "SessionColumnUserID" not "HumanUserIDCol"?

Comment (sp132) [reply internal/query/session.go:262]: You're absolutely right, I've glanced over it and noticed only `HumanUserIDCol` in the join statement

**Comment by sp132** [comment]
I'm confused with the linter output, not sure how to proceed with it. There are no number lines and I didn't even touch anything related to text or number comparison

`
  Error: singleCaseSwitch: should rewrite switch statement to if statement (gocritic)

  Error: missing cases in switch of type query.TextComparison: query.TextNotEquals, query.textCompareMax (exhaustive)

  Error: missing cases in switch of type query.NumberComparison: query.numberCompareMax (exhaustive)
`

Output of the failed integration test is very hard to read and so I'm not even sure what's the error is

**Comment by sp132** [comment]
I'd really appreciate some guidance on linter issues. It doesn't seem like these problems were introduced in this PR, so I need to understand the course of actions.

Do I somehow fix them in this PR?
D I ignore them?

**Comment by stebenz** [comment]
> I'd really appreciate some guidance on linter issues. It doesn't seem like these problems were introduced in this PR, so I need to understand the course of actions.
> 
> Do I somehow fix them in this PR? Do I ignore them?

The linter is configured so that only changed files get checked, and it seems that there were problems in this files before we used this configuration. To merge it they would need to be resolved, but I can try to help you with that.

1. "singleCaseSwitch: should rewrite switch statement to if statement (gocritic)"(internal/query/search_query.go#L447): change the switch to an if statement, that's about it
2. "missing cases in switch of type query.TextComparison: query.TextNotEquals, query.textCompareMax (exhaustive)" (internal/query/search_query.go#L245): adding a "case query.TextNotEquals: return sq.NotEq{q.Column.identifier(): q.Text}" and "default: return nil" should cover this
3. "missing cases in switch of type query.NumberComparison: query.numberCompareMax (exhaustive)"(internal/query/search_query.go#L346): adding a "default: return nil" should cover this

**Comment by sp132** [comment]
@stebenz thank you!

Had to add explicit switch cases for `textCompareMax` and `numberCompareMax` though. Default case wasn't satisfying enough for the linter.

**Review by stebenz — APPROVED** [review]

**Pull request merged**